data_processing/meteo/task_functions.py

```diff
@@ -186,60 +264,113 @@ def upload_files_datagouv(ti, minio_folder):
+                f"/resources/{resources_lists[path][url]}/"
+            )
+            r = requests.put(
+                url=resource_api_url,
+                json={},
+                headers={"X-API-KEY": DATAGOUV_SECRET_API_KEY},
+            )
+            r.raise_for_status()
+            continue
+        else:
+            # differenciation ressource principale VS documentation
+            is_doc = False
+            description = ""
+            try:
+                # two known errors:
+                # 1. files that don"t match either pattern
+                # 2. files that are contained in subfolders => creating new paths in config
+                if config[path]["doc_pattern"] and re.match(config[path]["doc_pattern"], file_with_ext):
+                    resource_name = file_with_ext.split(".")[0]
+                    is_doc = True
+                else:
+                    # peut-être mettre un 'if params else' ici pour publier les ressources
+                    # qui posent problème en doc quoiqu'il ?
+                    try:
+                        params = re.match(config[path]["source_pattern"], file_with_ext).groupdict()
+                    except AttributeError:
+                        print("File is not matching pattern")
+                        continue
+                    for k in params:
+                        # removing hook names for data.gouv display
+                        params[k] = clean_hooks(params[k])
+                    resource_name = config[path]["name_template"].format(**params)
+                    description = config[path]["description_template"].format(**params)
+                # if the resource doesn't exist yet, we simply create it
+                if clean_file_path in new_files:
+                    print(
+                        f"Creating new {'documentation ' if is_doc else ''}resource for:",
+                        file_with_ext
+                    )
+                    post_remote_resource(
+                        api_key=DATAGOUV_SECRET_API_KEY,
+                        remote_url=url,
+                        dataset_id=config[path]["dataset_id"][AIRFLOW_ENV],
+                        filesize=minio_files[file_path],
+                        title=resource_name if not is_doc else file_with_ext,
+                        type="main" if not is_doc else "documentation",
+                        format=get_file_extention(file_with_ext),
+                        description=description,
+                    )
+                    new_files_datasets.add(path)
+                    updated_datasets.add(path)
+                # resource is updated with a new name, we redirect the existing resource, rename it
+                # and update size and description
+                # values because we are dealing with the resource's new name
+                elif clean_file_path in files_to_update_new_name.values():
```

Comment:
We should look in files_to_update_new_name keys, and not values, since keys are new names.

```suggestion
                elif clean_file_path in files_to_update_new_name:
```